plugins/coverbrowser.koplugin/listmenu.lua

```diff
@@ -372,7 +372,10 @@ function ListMenuItem:update()
             if DocSettings:hasSidecarFile(self.filepath) then
                 self.been_opened = true
                 self.menu:updateCache(self.filepath, nil, true, pages) -- create new cache entry if absent
-                pages, percent_finished, status, has_highlight = unpack(self.menu.cover_info_cache[self.filepath])
+                pages            = self.menu.cover_info_cache[self.filepath][1]
+                percent_finished = self.menu.cover_info_cache[self.filepath][2]
+                status           = self.menu.cover_info_cache[self.filepath][3]
+                has_highlight    = self.menu.cover_info_cache[self.filepath][4]
```

Comment:
May be to avoid 4 similar lookups:
```lua
local cached_info = self.menu.cover_info_cache[self.filepath]
pages             = cached_info[1]
percent_finished  = cached_info[2]
...
```

(Also, I have a vague memory @NiLuJe may have had some other (shorter?) solution about intermediate nil values - does that ring a bell ?)

Reply:
That's it ;o). 

If the original table has been packed via `table.unpack`, it *can* contain intermediate `nil`s.

Reply:
c.f., https://github.com/koreader/koreader-base/pull/1535 for more details

Reply:
Looking at `CoverMenu:updateCache`, that would be perfectly applicable there ;).